zio/zngio/reader.go

```diff
@@ -95,9 +95,12 @@ func (r *Reader) LastSOS() int64 {
 func (r *Reader) ReadPayload() (*zng.Record, []byte, error) {
 again:
 	b, err := r.read(1)
-	if err == io.EOF || len(b) == 0 {
+	if err == io.EOF || err == peeker.ErrTruncated {
 		return nil, nil, nil
 	}
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
How about nesting the first if inside the second? And why does it make sense to treat peeker.ErrTruncated this way?

Reply:
ErrTruncated means that there is nothing left to read (after doing a peek(1) specifically.

Reply:
(in other words, it is like an io.EOF  in this case. i'll add a comment).